### ISSUE TITLE
SwiftLint should emit cursor output like compilers do

### ISSUE THREAD

**Issue by lilyball** [realm/SwiftLint#2268]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Enhancement Request

When SwiftLint emits a warning or error, it would be great if it could include the line of code and a cursor like compilers do.

For example, with the following Swift compiler output:

```
Untitled.swift:1:4: error: binary operator '+' cannot be applied to operands of type 'Int' and 'String'
23 + "foo"
~~ ^ ~~~~~
```

the `~~ ^ ~~~~~` is the cursor, and it's preceded by the line of code in question.

The reason I want this output is primarily because `xcpretty` apparently requires it in order to actually display warnings/errors. Without this, `xcpretty` is completely swallowing up all SwiftLint output.

**Comment by jpsim** [comment]
This is a great idea! Especially useful too if you're using SwiftLint from the command line and fixing violations in an editor other than Xcode, which I do from time to time.

If you have some time, this would be a fantastic contribution.

A good place to start would be to take a look at a similar function we use in our tests to render violations inline with the content. It's quite simple, it's called [`render(violations:in:)`](https://github.com/realm/SwiftLint/blob/0.28.2/Tests/SwiftLintFrameworkTests/TestHelpers.swift#L62-L80) and outputs violations in this format:

```swift
func abc()↓{
          ^ warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
}
```

**Comment by lilyball** [comment]
I'd still really love to see this, though I don't have time to work on it myself.
